eventsourcing/src/main/java/org/axonframework/eventsourcing/configuration/SimpleEventSourcedEntityModule.java

```diff
@@ -0,0 +1,219 @@
+/*
+ * Copyright (c) 2010-2025. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.eventsourcing.configuration;
+
+import jakarta.annotation.Nonnull;
+import org.axonframework.commandhandling.CommandBus;
+import org.axonframework.commandhandling.CommandHandlingComponent;
+import org.axonframework.common.BuilderUtils;
+import org.axonframework.common.FutureUtils;
+import org.axonframework.configuration.BaseModule;
+import org.axonframework.configuration.ComponentBuilder;
+import org.axonframework.configuration.ComponentDefinition;
+import org.axonframework.configuration.Configuration;
+import org.axonframework.configuration.LifecycleRegistry;
+import org.axonframework.eventsourcing.CriteriaResolver;
+import org.axonframework.eventsourcing.EventSourcedEntityFactory;
+import org.axonframework.eventsourcing.EventSourcingRepository;
+import org.axonframework.eventsourcing.eventstore.EventStore;
+import org.axonframework.lifecycle.Phase;
+import org.axonframework.modelling.StateManager;
+import org.axonframework.modelling.command.EntityIdResolver;
+import org.axonframework.modelling.entity.EntityCommandHandlingComponent;
+import org.axonframework.modelling.entity.EntityModel;
+import org.axonframework.modelling.repository.Repository;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Basis implementation of the {@link EventSourcedEntityModule}.
+ *
+ * @param <I> The type of identifier used to identify the event-sourced entity that's being built.
+ * @param <E> The type of the event-sourced entity being built.
+ * @author Steven van Beelen
+ * @since 5.0.0
+ */
+class SimpleEventSourcedEntityModule<I, E>
```

Comment:
We decided for `ID` instead of `I` for the identifier...right? :sweat_smile: 
I am beginning to doubt now in all honesty.
Anyhow, feel free to rename the generic. The IDE should help you with that.